src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/domainevents/DomainEventDescriber.kt

```diff
@@ -87,13 +88,19 @@ class DomainEventDescriber(
     return event.describe { "The assessment was appealed and ${it.eventDetails.decision.value}. The reason was: ${it.eventDetails.decisionDetail}" }
   }
 
-  private fun buildPlacementApplicationWithdrawnDescription(domainEventSummary: DomainEventSummary): String {
+  private fun buildPlacementApplicationWithdrawnDescription(domainEventSummary: DomainEventSummary): String? {
     val event = domainEventService.getPlacementApplicationWithdrawnEvent(domainEventSummary.id())
-    val dates = event?.data?.eventDetails?.placementDates ?: emptyList()
-    return "A request for placement was withdrawn" +
-      if (dates.isNotEmpty()) {
-        " for dates " + dates.joinToString(", ") { "${it.startDate.toUiFormat()} to ${it.endDate.toUiFormat()}" }
-      } else { "" }
+
+    return event.describe { data ->
+      val dates = data.eventDetails.placementDates ?: emptyList()
+      val reasonDescription = data.eventDetails.withdrawalReason.javaConstantNameToSentence()
+
+      "A request for placement was withdrawn" +
+        if (dates.isNotEmpty()) {
+          " for dates " + dates.joinToString(", ") { "${it.startDate.toUiFormat()} to ${it.endDate.toUiFormat()}" }
+        } else { "" } +
+        ". The reason was $reasonDescription"
```

Comment:
I think I might like to see a colon here - so `The reason was: $reasonDescription`